mono/utils/mono-log-windows.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * mono-log-windows.c: Simplistic simulation of a syslog logger for Windows
+ *
+ * This module contains the Windows syslog logger interface
+ *
+ * Author:
+ *    Neale Ferguson <neale@sinenomine.net>
+ *
+ */
+#include <config.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+
+#ifdef HOST_WIN32
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+#include <glib.h>
+#include <errno.h>
+#include <time.h>
+#include "mono-logger.h"
+
+static FILE *logFile = NULL;
+static void *logUserData = NULL;
+static char *logFileName = L".//mono.log";
+
+/**
+ * mapSyslogLevel:
+ * 	
+ * 	@level - GLogLevelFlags value
+ * 	@returns The equivalent character identifier
+ */
+static inline char 
+mapLogFileLevel(GLogLevelFlags level) 
+{
+	if (level & G_LOG_LEVEL_ERROR)
+		return ('E');
+	if (level & G_LOG_LEVEL_CRITICAL)
+		return ('C');
+	if (level & G_LOG_LEVEL_WARNING)
+		return ('W');
+	if (level & G_LOG_LEVEL_MESSAGE)
+		return ('N');
+	if (level & G_LOG_LEVEL_INFO)
+		return ('I');
+	if (level & G_LOG_LEVEL_DEBUG)
+		return ('D');
+	return ('I');
+}
+
+/**
+ * mono_log_open_logfile
+ * 	
+ *	Open the logfile. If the path is not specified default to stdout. If the
+ *	open fails issue a warning and use stdout as the log file destination.
+ *
+ * 	@ident - Identifier: ignored
+ * 	@userData - Not used
+ */
```

Comment:
Fixed

From: Johan Lorensson +ADw-notifications+AEA-github.com+ADw-mailto:notifications+AEA-github.com+AD4APg-
Reply-To: mono/mono +ADw-reply++0004b6fa1d2a9905b1457be4bd84719c4f162e57e2f8102a92cf00000001136902dc92a170ce03ea18AQA-reply.github.com+ADw-mailto:reply++0004b6fa1d2a9905b1457be4bd84719c4f162e57e2f8102a92cf00000001136902dc92a170ce03ea18AQA-reply.github.com+AD4APg-
Date: Friday, June 3, 2016 at 4:33 AM
To: mono/mono +ADw-mono+AEA-noreply.github.com+ADw-mailto:mono+AEA-noreply.github.com+AD4APg-
Cc: Neale Ferguson +ADw-neale+AEA-sinenomine.net+ADw-mailto:neale+AEA-sinenomine.net+AD4APg-, Mention +ADw-mention+AEA-noreply.github.com+ADw-mailto:mention+AEA-noreply.github.com+AD4APg-
Subject: Re: +AFs-mono/mono+AF0- Enhance log tracing (+ACM-2734)


In mono/utils/mono-log-windows.c+ADw-https://github.com/mono/mono/pull/2734+ACM-discussion+AF8-r65673258+AD4-:

+AD4-       if (level +ACY- G+AF8-LOG+AF8-LEVEL+AF8-INFO)
+AD4-               return ('I')+ADs-
+AD4-       if (level +ACY- G+AF8-LOG+AF8-LEVEL+AF8-DEBUG)
+AD4-               return ('D')+ADs-
+AD4-       return ('I')+ADs-
+AD4- +AH0-
+AD4-
+AD4- +ACoAKg-
+AD4-  +ACo- mono+AF8-log+AF8-open+AF8-logfile
+AD4-  +ACo-
+AD4-  +ACo-    Open the logfile. If the path is not specified default to stdout. If the
+AD4-  +ACo-    open fails issue a warning and use stdout as the log file destination.
+AD4-  +ACo-
+AD4-  +ACo-    +AEA-ident - Identifier: ignored
+AD4-  +ACo-    +AEA-userData - Not used
+AD4-  +ACo-/


Comment doesn't match function, copy paste?

+IBQ-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub+ADw-https://github.com/mono/mono/pull/2734/files/583d1a2580a5fc61d0002920c17c23fa71375c3d+ACM-r65673258+AD4-, or mute the thread+ADw-https://github.com/notifications/unsubscribe/AAS2-sJyQU7A8IGlkY14d9IiH7ibZOxNks5qH-bcgaJpZM4HsGtY+AD4-.